homeassistant/components/recorder/db_schema.py

```diff
@@ -122,6 +123,8 @@ def result_processor(self, dialect, coltype):  # type: ignore[no-untyped-def]
     .with_variant(postgresql.DOUBLE_PRECISION(), "postgresql")
 )
 
+TIMESTAMP_TYPE = Float(16)  # Covers up to at least 2250 AD
```

Comment:
We could also use Decimal so we don't have any issues with exact values but that would take more space and likely isn't worth the trade off

Reply:
Looks like we use `DOUBLE_TYPE` for statistics so lets do the same here